docs/Testing.md

```diff
@@ -10,12 +10,156 @@ This isn't a hard and fast rule, but it does express our approach fairly well.
+
+```ts
+it("given a date in the past, colorForDueDate() returns red", () => {
+  const input = colorForDueDate("2000-10-20")
+  expect(input).toBe("red")
+})
+```
+
+Jest functions like `it`, `test`, `expect`, and more are loaded globally by the Jest test runner, so you don't need to import them.
+
+### Best Practices
+
+> When writing a test, do your best to make sure that your tests > include the following information:
+>
+> - Given - some precondition
+> - When - some action executed by the function that you’re testing
+> - Then - the expected outcome
+>   This is also known as AAA (Arrange, Act, Assert).
+
+-- <cite>[Structuring Tests in React Native Docs](https://reactnative.dev/docs/testing-overview#structuring-tests)</cite>
+
+You can read more about how to best practices for creating tests in the [Structuring Tests](https://reactnative.dev/docs/testing-overview#structuring-tests) section of the React Native docs.
+
+### Writing Tests
+
+To write your own tests, create a `.test.ts` file within the `app` or `test` directory.
+
+Then run `yarn test` to run all unit tests using Jest.
+
+When writing tests, you can also run Jest in watch mode by running `yarn test:watch`. This will start a long running Jest process, that re-runs your tests on save in your editor. This is useful for iterating on values and getting quick feedback about whether your changes were successful or not.
+
+### When to write Unit Tests
+
+Some developers believe that every line of code should have a co-responding test. However, Infinite Red believes that some unit tests provide more values than others. Here are a few situations where we have found unit tests to be helpful in client projects:
+
+- **complicated regex's**: for many developers, regex's only make sense as you are writing them. Testing with a series of valid and invalid inputs can help ensure that they work as intended for future developers.
+- **nested if/else statements**: if you rely heavily on a function with a lot of if/else statements, it can be helpful to have tests to make sure that you can visit each condition. Often times when we have more than a handful of conditionals, it can become impossible to visit all of them without realizing it.
+- **validation functions**: often times we may write functions like `isJson()` to validate that a value is a specific shape. If critical parts of our code rely on the correctness of this function, it can be useful to add tests to make sure that it fails on invalid inputs and passes on valid inputs.
 
 ## Mocking
 
-MAVERICKTODO: Add info about mocking JS & native here.
+However, often our code is not entirely written as pure functions. Most apps have side effects like making network requests, calling native modules, or reaching for global objects.
+
+Relying on external code can make our code difficult and slow to test.
+
+One solution to making our tests easier to write is by utilizing mocks.
```

Comment:
We say "one solution" but then don't enumerate anything else. I think it's important to mention somewhere here that if code's functionality relies on external systems, it may be more appropriate to test that code with E2E tests, rather than using mocking.

Reply:
I've updated that phrasing here: https://github.com/infinitered/ignite/pull/2196/commits/df5220851f804656deaa273253f6640b537ac360